client/lib/analytics/page-view-tracker/index.jsx

```diff
@@ -44,12 +48,21 @@ export class PageViewTracker extends React.Component {
 		clearTimeout( this.state.timer );
 	}
 
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.path !== this.props.path ||
+			prevProps.selectedSiteId !== this.props.selectedSiteId
+		) {
+			this.queuePageView();
+		}
+	}
+
 	queuePageView = () => {
-		const { delay = 0, path, recorder = noop, title } = this.props;
+		const { delay = 0, path, recorder = noop, sitesLoaded, title } = this.props;
 
 		debug( `Queuing Page View: "${ title }" at "${ path }" with ${ delay }ms delay` );
 
-		if ( this.state.timer ) {
+		if ( ! sitesLoaded || this.state.timer ) {
```

Comment:
I wonder if we could update this check to instead verify that if the current path has a `siteSlug`, do we have a valid `selectedSiteId`?

Reply:
Anothter option: If the path has a slug, maybe use `getSiteSlug( selectedSiteId )` to see if it matches the slug provided in the path?